src/code_f8f60_len_1560.c

```diff
@@ -2,23 +2,34 @@
 
 s32 MakeLerp(script_context* script, s32 initialCall) {
     s32* ptrReadPos = script->ptrReadPos;
-    s32* ptrNextPos = *ptrReadPos++;
-    s32* ptrNextPos2;
-    s32* ptrNextPos3;
-    s32* ptrNextPos4;
-
-    script->varTable[12] = get_variable(script, ptrNextPos);
-    ptrNextPos2 = *ptrReadPos++;
-    script->varTable[13] = get_variable(script, ptrNextPos2);
-    ptrNextPos3 = *ptrReadPos++;
-    script->varTable[15] = get_variable(script, ptrNextPos3);
-    ptrNextPos4 = *ptrReadPos++;
-    script->varTable[11] = get_variable(script, ptrNextPos4);
-    script->varTable[14] = 0;
+
+    script->varTable[0xC] = get_variable(script, *ptrReadPos++); // start
+    script->varTable[0xD] = get_variable(script, *ptrReadPos++); // end
+    script->varTable[0xF] = get_variable(script, *ptrReadPos++); // duration
+    script->varTable[0xB] = get_variable(script, *ptrReadPos++); // easing type
+    script->varTable[0xE] = 0; // elapsed
+
     return 2;
 }
 
-INCLUDE_ASM(code_f8f60_len_1560, UpdateLerp);
+s32 UpdateLerp(script_context* script, s32 initialCall) {
+    script->varTable[0x0] = (s32) update_lerp(
+        script->varTable[0xB],
+        (s32) script->varTable[0xC],
```

Comment:
I'm not sure these casts are necessary since these variables are already s32s

Reply:
They're necessary! `update_lerp` takes `f32` in these two slots, and without the cast the compiler attempts to convert the s32 ints to float rather than just moving them to f registers (not converting).

Reply:
wait no